src/com/cft/contactmerge/FirstAndLastNamesMatchLogic.java

```diff
@@ -8,18 +11,60 @@
             {"Al", "Alan"},
             {"Bob", "Robert"}
     };
+    private final static String[] namePunctuations = {".", ",", "'", "-"};
+
+    /***********************************************
+     * Method(s) used to call specific comparisons *
+     ***********************************************/
 
-    public static AnswerType doFirstAndLastNamesMatch(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+    public static AnswerType doFirstAndLastNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
         return getMatchingResultFromMatchingMethodsList(contactOneFirstName, contactOneLastName, contactTwoFirstName, contactTwoLastName);
     }
 
+    /*************************************
+     * Use all logical comparison method *
+     *************************************/
+
     private static AnswerType getMatchingResultFromMatchingMethodsList(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
         // Do first and last names have an exact match (case sensitive)? If so, return yes.
-        if (isFirstAndLastNamesExactMatchCaseSensitive(contactOneFirstName, contactOneLastName, contactTwoFirstName, contactTwoLastName)){
+        if (isMatchMatchCaseSensitive(contactOneFirstName, contactOneLastName, contactTwoFirstName, contactTwoLastName)){
             return AnswerType.yes;
         }
 
         // Do first and last names have an exact match (case Insensitive)? If so, return yes.
+        if (isMatchCaseInsensitive(contactOneFirstName, contactOneLastName, contactTwoFirstName, contactTwoLastName)){
```

Comment:
Doing a separate test for isMatchCaseInsensitive() and isMatchWithWhiteSpaces() is unnecessary. They are already accomplished by your isMatchWithWhiteSpaces_CaseInsensitive(). An exact match with spaces and case checking isn't "more" AnswerType.yes. In fact, I think you should even be able to perform all of them with a call to isMatchWithPunctuations() if you modify isMatchWithPunctuations() to do a case insensitive compare. Less is more.

Reply:
Yup, as discussed yesterday, I am going to refactor the code to use very little methods. I will use a string formatted method to change strings for matching, so very little methods will be used.